modules/Core/src/main/java/org/terasology/core/world/generator/ClimateSimulatorData.java

```diff
@@ -0,0 +1,131 @@
+package org.terasology.core.world.generator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Encapsulates climate distance calculations for the climate simulator 
+ * 
+ * @author Michael
+ *
+ */
+public class ClimateSimulatorData {
+	private static final Logger logger = LoggerFactory.getLogger(ClimateSimulator.class);
+	private float[][] heightmap;
+	private int size;
+
+	public ClimateSimulatorData(float[][] heightmap, int size) {
+		this.heightmap = heightmap;
+		this.size = size;
+	}
+	
+	public float[][] getHeightmap() {
+		return heightmap;
+	}
+	
+	public int getSize() {
+		return size;
+	}
+	
+	public float[][] initDist(String fromWhat) {
```

Comment:
Could this refactoring go further and split into individual methods for each "fromWhat" value? That would avoid the if (which alternatively could be turned into a switch)